### PR TITLE
Add NEWS entry for #9020.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 This file contains a summary of important user-visible changes.
 
+**Changes**
+
+- API: Sort, Term, Op and datatype objects do not maintain a reference
+       (and depend) on Solver anymore. Instead, they are now associated
+       to a NodeManager object. This allows sharing of NodeManagers between
```
Comment (ajreynol) [inline NEWS.md:7]:
```suggestion
       to a NodeManager object. This allows sharing of these objects between
```

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 This file contains a summary of important user-visible changes.
 
+**Changes**
+
+- API: Sort, Term, Op and datatype objects do not maintain a reference
```
Comment (ajreynol) [inline NEWS.md:5]: Also add `Grammar` to this list.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: NEWS.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
 This file contains a summary of important user-visible changes.
 
+**Changes**
+
+- API: Sort, Term, Op, Grammar and datatype objects do not maintain a reference
```
Comment (ajreynol) [inline NEWS.md:5]: Actually, do we even need to mention NodeManager here at all?  The user shouldnt have to care about what NodeManager is.

Comment (aniemetz) [reply NEWS.md:5]: Well, yes, technically the only thing important is that we can now share NodeManager instances between Solvers.

Comment (aniemetz) [reply NEWS.md:5]: For us, however, it might be good to have these details added? Idk, don't have a strong opinion.

Comment (ajreynol) [reply NEWS.md:5]: Actually, we have always shared NodeManagers between Solvers, its only the dependency that has changed.   So the wording is still confusing to me.

I think I'd prefer not mentioning the internals in this note.

Comment (aniemetz) [reply NEWS.md:5]: Right, we have always shared NodeManagers between Solvers, but until #9020 we were not able to share Terms, Sorts, etc. between Solver instances. I agree, let's not mention internals, and only mention this.

File: NEWS.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
 This file contains a summary of important user-visible changes.
 
+**Changes**
+
+- API: Previosuly, it was disallowed to share Sort, Term, Op, Grammar and
```
Comment (ajreynol) [inline NEWS.md:5]:
```suggestion
- API: Previously, it was not possible to share Sort, Term, Op, Grammar and
```

**Review by ajreynol — CHANGES_REQUESTED** [review]

**Review by ajreynol — APPROVED** [review]

**Pull request merged**